app/src/main/java/org/simple/clinic/registration/name/RegistrationNameEffectHandler.kt

```diff
@@ -19,7 +21,16 @@ class RegistrationNameEffectHandler @AssistedInject constructor(
   fun build(): ObservableTransformer<RegistrationNameEffect, RegistrationNameEvent> {
     return RxMobius
         .subtypeEffectHandler<RegistrationNameEffect, RegistrationNameEvent>()
-        .addConsumer(PrefillFields::class.java, { uiActions.preFillUserDetails(it.entry)}, schedulers.ui())
+        .addConsumer(PrefillFields::class.java, { uiActions.preFillUserDetails(it.entry) }, schedulers.ui())
+        .addTransformer(ValidatedEnteredName::class.java, validateNameEntry())
         .build()
   }
+
+  private fun validateNameEntry(): ObservableTransformer<ValidatedEnteredName, RegistrationNameEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .map { if (it.name.isBlank()) Blank else Valid }
```

Comment:
One question I have is when do we consider doing validations in effect handler vs update function. Because I remember previously we have done some validations in update functions. Since this is a simple validation should we have done this in update function or am I missing something? 🤔

Reply:
You're right. I am planning to do an entire sweep of the entire registration flow once I migrate all the screens to Mobius. Unifying the way we do validations here is one of them.

I will make this change then.